pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial.md

```diff
@@ -0,0 +1,283 @@
+    });
+}; 
+// is called when user types in input
+element.addEventListener("keyup", checkPassword);
+// is called if user pastes into input 
+element.addEventListener("change", checkPassword);  
+```
+
+
+Inside the `checkPassword` function, we’ll call two more:
+
+
+
+*   If the password passes a defined check, it will call `checkMet`, turning the font green. 
+*   If the password failed a defined check, it will call `checkRemoved`, turning the font red.
+
+```js
+function initCheckPassword(element) {
+  // removes invalid class and replaced with valid if check is met
+  const checkMet = (listItem) => {
+    listItem.classList.remove("checkFail");
+    listItem.classList.add("checkPass");
+  };
+  // removes valid class and replaced with valid if check is deleted
+  const checkRemoved = (listItem) => {
+    listItem.classList.remove("checkPass");
+    listItem.classList.add("checkFail");
+  }
+  // function goes over each check to see if its been met
+  const checkPassword = () => {
+    let successTest = passwordChecks.forEach((item) => {
+      let passed = item.checkRegEx(element.value);
+      // calls checkMet or checkRemoved
+        passed ? checkMet(item.text) : checkRemoved(item.text);
+    });
+ }
+    // is called when user types in input
+    element.addEventListener("keyup", checkPassword);
+    // is called if user pastes into input 
+    element.addEventListener("change", checkPassword);
+};
+```
+
+
+
+Both of our check functions add and remove classes from the password requirement list items within the `rules` div. The `checkFailed` class will make our text red and the `checkPassed` will change it to green. 
+
+Just like non-AMP pages, once the `amp-script` logic adds or removes a class it will be reflected in the DOM. These classes are treated the same as all other CSS that don't work with `amp-script`. 
+
+For this to work, we will need to add the `checkFail` class to all the items and define both `checkPass` and `checkFailed` within the `<style amp-custom>` tag in the page head. 
+
+
+```html
+<ul>
+    <li id="lowercase" class="checkFail">Lowercase letter</li>
+    <li id="capital" class="checkFail">Capital letter</li>
+    <li id="number" class="checkFail">Number</li>
+    <li id="special" class="checkFail">Special character (@$!%*?&)</li>
+    <li id="eight" class="checkFail">At least 8 characters long</li>
+</ul> 
+```
+
+
+
+```css
+.checkPass {
+    color: #2d7b1f;
+} 
+.checkFail {
+    color:#c11136;
+}
+```
+
+
+Refresh the page and type into the password input. The elements corresponding the check passed or failed should turn red or green accordingly!
+
+<figure class="alignment-wrapper margin-">
+  <amp-video layout="intrinsic" poster="/static/img/docs/guides/storiesbp/do-background-still.jpg" width="400" height="720" loop autoplay noaudio>
+    <source src="/static/img/docs/tutorials/custom-javascript-tutorial/checkpassfail.mp4" type="video/mp4" />
+  </amp-video>
+</figure>
+
+Our logic is now complete!
```

Comment:
You might want to explicitly mention the explicit call to `initCheckPassword` to setup the event handlers.